pkg/capr/planner/config.go

```diff
@@ -136,7 +136,7 @@ func addRoleConfig(config map[string]interface{}, controlPlane *rkev1.RKEControl
 	// to run.
 	if isControlPlane(entry) {
 		logrus.Debug("addRoleConfig rendering arguments and mounts for kube-controller-manager")
-		certDirArg, certDirMount := renderArgAndMount(config[KubeControllerManagerArg], config[KubeControllerManagerExtraMount], runtime, DefaultKubeControllerManagerDefaultSecurePort, DefaultKubeControllerManagerCertDir)
+		certDirArg, certDirMount := renderArgAndMount(controlPlane.Spec.MachineGlobalConfig.Data[KubeControllerManagerArg], controlPlane.Spec.MachineGlobalConfig.Data[KubeControllerManagerExtraMount], runtime, DefaultKubeControllerManagerDefaultSecurePort, DefaultKubeControllerManagerCertDir)
```

Comment:
Why was this done? The previous function `addUserConfig` is supposed to add `MachineGlobalConfig` and `MachineSelectorConfig` to `config`, and then filter out non KDM data. Were you seeing data filtered? If so it should likely be updated in KDM.

Reply:
actually I did not have much experience with rke2 ...just saw that this test is broken on 2.8 head also...but it is passing in CI not sure how....so what I did is I ran the test and then tried to fix the test problems.....

For this particular thing I was getting error because even though I was setting machine global config in testdata the directory value was not right...so first I changed the machine global config key value to array of string in test data which should ideally be the case and then made this change to read these values from machinGlobal config first and use defaults otherwise.

Maybe my understanding is wrong in some cases....please let me know what should be the value here ... because the existing code breaks the test